From 906eb84f9098167fc82739ad672cbab215778563 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 7 Jun 2022 17:41:52 +0800
Subject: [PATCH] sw64: remove MAX_ASN
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

There's no need to manually define MAX_ASN. Use calculated
HARDWARE_ASN_MASK instead.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/mmu_context.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/sw_64/include/asm/mmu_context.h b/arch/sw_64/include/asm/mmu_context.h
index 6b2ab3224ec9..e3d7ae7c873e 100644
--- a/arch/sw_64/include/asm/mmu_context.h
+++ b/arch/sw_64/include/asm/mmu_context.h
@@ -48,7 +48,6 @@ __reload_thread(struct pcb_struct *pcb)
  */
 
 #ifdef CONFIG_SUBARCH_C3B
-#define MAX_ASN			1023
 #define WIDTH_HARDWARE_ASN	10
 #endif
 
@@ -89,7 +88,7 @@ __get_new_mm_context(struct mm_struct *mm, long cpu)
 	unsigned long asn = cpu_last_asn(cpu);
 	unsigned long next = asn + 1;
 
-	if ((asn & HARDWARE_ASN_MASK) >= MAX_ASN) {
+	if ((asn & HARDWARE_ASN_MASK) >= HARDWARE_ASN_MASK) {
 		tbiap();
 		next = (asn & ~HARDWARE_ASN_MASK) + ASN_FIRST_VERSION;
 	}
-- 
2.34.1

